src/tests/Interop/common/xplatform.h

```diff
@@ -8,6 +8,106 @@
 
 #ifndef WINDOWS
 
+#define __RPC_FAR
+#define DECLSPEC_UUID(x)
+#define DECLSPEC_NOVTABLE
+#define MIDL_INTERFACE(x)   struct DECLSPEC_UUID(x) DECLSPEC_NOVTABLE
+#ifndef OBJC_TESTS
+#define interface struct
+#endif
+#define STDMETHOD(method)       virtual HRESULT STDMETHODCALLTYPE method
+#define STDMETHOD_(type,method) virtual type STDMETHODCALLTYPE method
+#undef _In_
+#define _In_
+#undef _Outptr_
+#define _Outptr_
+#undef _Out_
+#define _Out_
+#undef _In_opt_
+#define _In_opt_
+#undef _COM_Outptr_
+#define _COM_Outptr_
+#undef _Inout_
+#define _Inout_
+#define __RPC__out
+#define __RPC__in
+#define __RPC__deref_out
+#define __RPC__deref_out_opt
+#define __RPC__deref_out_ecount_full_opt(x)
+#define __RPC_unique_pointer
+
+
+#define E_POINTER                        _HRESULT_TYPEDEF_(0x80004003L)
```

Comment:
This is inconsistent with `E_NOTIMPL`. Why do we use `_HRESULT_TYPEDEF_` for some and not others? Instead I would just cast if necessary otherwise follow the `E_NOTIMPL` pattern below.

Reply:
ok